### DISCUSSION TITLE
Fail to install from source

### DISCUSSION THREAD

**Discussion by qqlearn123** [helix-editor/helix#8401] [Troubleshooting]
When I install from latest repo, I encountered the following errors:
```
Failure 1/3: blueprint Git command failed.
Stout:
Sterr: fatal: unable to access 'https://gitlab.com/Gambia/tree-sitter-blueprint/': Proxy CONNECT aborted

Failure 2/3: wren Git command failed.
Stdout:
Stderr: error: Server does not allow request for unadvertised object

Failure 3/3: t32 Git command failed.
Stdout:
Stderr: error: RPC failed; result=22, HTTP code = 422
fatal: the remote end hung up unexpectedly

--- stderr
thread 'main' panicked at helix-term/build.rs:5:26:
Failed to fetch tree-sitter grammars: 3 grammars failed to fetch
```

Not sure what is the cause for the errors. Thanks for any help!

**Comment by pascalkuthe** [comment]
Your git version is too old

**Comment by qqlearn123** [comment]
> Your git version is too old

I have git v2.40 which should not be too old? If it is old, do you know a version of the repo compatible with git v2.40?

Thanks!

**Comment by Abraxas-Knister** [comment]
If git being too old is the issue, why doesn't the installer tell me so, and do so before it wastes any time? And secondly, git v2.40 was released this year. Why on earth would it make sense for helix as a project to have already deprecated it?

**Reply by qqlearn123** [reply]
Agree

**Comment by the-mikedavis** [comment]
To me these errors look like random network failures. Git v2.40 should be up-to-date-enough. I'm not sure which git version introduced the feature exactly but we clone these grammar repositories at exact revisions, so we need a git version that supports that operation. Some old debian/ubuntu installations have git versions that are too old and we get similar-looking errors (see https://github.com/helix-editor/helix/issues/7036).

For this case I think you should just retry the build. If those specific grammars consistently fail then you can remove them from the build:

```toml
# ~/.config/helix/languages.toml
use-grammars = { except = ["wren", "t32", "blueprint"] }
```

**Reply by qqlearn123** [reply]
Thanks. Is there any way to install grammars manually myself so that I can just use pre-downloaded files without encountering network error.

**Reply by the-mikedavis** [reply]
If the build script sees a `HELIX_DISABLE_AUTO_GRAMMAR_BUILD` environment variable set it won't try to fetch and build the grammars. Then you can use `hx -g fetch` and `hx -g build` afterwards to build these separately. You could also manually clone the repositories into `runtime/grammars/sources/<grammar name>` at the revisions in the `languages.toml`. When building, if the revision in that directory matches what we have in the `languages.toml` then we don't try to use the network at all.

**Reply by Disonantemus** [reply]
I had the same problem today, I'm no programmer:

- Please put that in the [Installation Documentation](https://docs.helix-editor.com/install.html)
- [`flake.nix`](https://github.com/helix-editor/helix/blob/3915b04bd91084477b3076952e6ad6cfdd414e72/flake.nix#L130) has it

**Installed with:**

~~~ocaml
export HELIX_DISABLE_AUTO_GRAMMAR_BUILD=1
cargo install --path helix-term --locked
~~~

**Then:**
~~~ocaml
$ hx --grammar fetch
Fetching 179 grammars
176 up to date git grammars
Failure 1/3: lpf Git command failed.
Stdout:
Stderr: fatal: unable to access 'https://gitlab.com/TheZoq2/tree-sitter-lpf/': The requested URL returned error: 403

Failure 2/3: t32 Git command failed.
Stdout:
Stderr: fatal: unable to access 'https://gitlab.com/xasc/tree-sitter-t32/': The requested URL returned error: 403

Failure 3/3: blueprint Git command failed.
Stdout:
Stderr: fatal: unable to access 'https://gitlab.com/gabmus/tree-sitter-blueprint/': The requested URL returned error: 403

Error: 3 grammars failed to fetch
~~~

---

**Setup:**
~~~ocaml
distro: Arch Linux x86_64
kernel: 6.1.52-1-lts
shell: bash 5.2.26
cpu: Intel i7-4790 (8) @ 3.600GHz
gpu: AMD ATI Radeon RX 470/480/570/570X/580/580X/590
~~~
